src/service.rs

```diff
@@ -61,6 +61,16 @@ impl ServiceRequest {
         ServiceRequest { req, payload }
     }
 
+    /// Get a mutable reference to parts
+    pub fn parts_mut(&mut self) -> (&mut HttpRequest, &mut Payload) {
```

Comment:
This should not be allowed, otherwise request could be cloned and you would get panic trying to modify request head